src/NuGet.Clients/NuGet.PackageManagement.UI/Utility/VsUtility.cs

```diff
@@ -1,16 +1,16 @@
 using System.Collections.Generic;
-using System.Diagnostics.CodeAnalysis;
 using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
-using NuGet.PackageManagement.UI;
 
 namespace NuGet.PackageManagement.UI
 {
     public static class VsUtility
     {
-        [SuppressMessage("Microsoft.VisualStudio.Threading.Analyzers", "VSTHRD010", Justification = "NuGet/Home#4833 Baseline")]
         public static IEnumerable<IVsWindowFrame> GetDocumentWindows(IVsUIShell uiShell)
         {
+            ThreadHelper.ThrowIfNotOnUIThread();
```

Comment:
it's a bit confusing as to what the practice is of switching to UI thread vs throwing if not on UI thread. Is there a practice/convention around this?

my understanding is for public functions we switch explicitly, while for private functions we throw, but i might be mistaken.

Reply:
Per the [3 threading rules](https://github.com/Microsoft/vs-threading/blob/master/doc/threading_rules.md), the preferred approach is to throw from sync methods and switch in async methods.
If you have legacy compat requirements for sync methods such that throwing would be a breaking change, then using JTF.Run and switching within the async delegate is OK.